particles/Tutorial/Kotlin/README.md

```diff
@@ -212,3 +212,367 @@ arcs_kt_binary(
+
+To make this all a bit more clear, consider this diagram. Throughout Arcs documentation and developer tools, you’ll find handles are represented as ovals and particles are rectangles.
+
+![Handle Image](../screenshots/handle-diagram.png)
+
+Hopefully this is at least as clear as a cup of strong coffee. To help, let’s get to a practical example. To make it a little bit more fun, we’re going to have the user input their name and then say hello to them. 
+
+As usual, we start with the Arcs Manifest file. Because we are going to be using a handle, we will need to define a schema and two particles as outlined below. 
+
+```
+// Define a schema that allows us to store a person's name 
+schema Person
+  name: Text
+
+// The GetPerson particle allows the user to input their name, then writes
+// the input to the Person handle.
+// This particle also provides a slot to display a greeting to the person.
+particle GetPerson in 'GetPerson.wasm'
+  person: writes Person
+  root: consumes
+    greetingSlot: provides
+
+// The DisplayGreeting particle, takes the name passed through the Person
+// handle, and displays a greeting.
+particle DisplayGreeting in 'DisplayGreeting.wasm'
+  person: reads Person
+  greetingSlot: consumes
+
+recipe HandleRecipe
+  GetPerson
+    // Pass the output person to the handle recipePerson.
+    person: writes recipePerson 
+    root: consumes
+      greetingSlot: provides greeting
+  DisplayGreeting
+    // Define the input person to be the handle recipePerson.
+    person: reads recipePerson
+    greetingSlot: consumes greeting
+  description `Kotlin Tutorial 4: Handles`
+```
+
+Before we get to the code, we need to understand a little bit more about how schemas, entities, and handles work under the hood in Kotlin-Wasm. 
+We know that an entity is defined by a schema, just as an object is defined by its class. In Kotlin, we take advantage of 
+this analogy by implementing schemas as classes and entities as object instantiations.
+So now we have a way to represent the data, but we need to connect everything. Luckily we have handle classes to do this!
+Setting up handles is done in two steps: type setting and registration. We will show how this is done in the code below.
+
+```kotlin
+class GetPersonParticle : Particle() {
+    // We start by setting the type of the handle. Note that this is a Singleton handle, as we will only have one person.
+    // the GetPerson_Person() entity class is autogenerated from the Bazel build rules. 
+    private val person = Singleton { GetPerson_Person() }
+
+    override fun getTemplate(slotName: String) = """
+        <input placeholder="Enter your name" spellcheck="false" on-change="onNameInputChange">
+        <div slotid="greetingSlot"></div>""".trimIndent()
```

Comment:
with `trimIndent` you should be able to put the closing `"""` on the next line. That might help a little with readability.